### PR TITLE
THIRDEYE-850 : Task driver in distributed anomaly detection should be able to run different types of tasks

### PR DESCRIPTION
**Pull request by npawar**
This pull request contains code to handle different task types in the workers. It also contains some additional refactoring.

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/task/TaskConstants.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package com.linkedin.thirdeye.anomaly.task;
+
+public class TaskConstants {
+
+  public enum TaskType {
```
Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/task/TaskConstants.java]: TaskExecutorType ?

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/job/JobContext.java [hunk 2]
```diff
@@ -33,7 +31,7 @@
 
   private long jobExecutionId;
 
-  public ThirdEyeJobContext() {
+  public JobContext() {
 
   }
```
Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/job/JobContext.java]: Nit : No need to add default c'tor.

**Comment by puneetjaiswal** [comment]
LGTM with minor comments.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/task/TaskConstants.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    AnomalyDetection {
+
+      @Override
+      Class<?> getKlazz() {
+        return AnomalyDetectionTaskRunner.class;
```
Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/task/TaskConstants.java]: Providing executor object for a task type should not be responsibility of TaskType. There should be a JobExecutorFactory which should proved JobExecutor based on TaskType.

Comment (npawar) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/task/TaskConstants.java]: Fixed

**Comment by npawar** [comment]
Created tasks of different types and tested that the workers pick them up and execute the corresponding function

**Pull request merged**